prescyent/dataset/motion/teleop_icub/dataset.py

```diff
@@ -0,0 +1,117 @@
+from prescyent.dataset.motion.dataset import MotionDataset, Episodes
+from prescyent.utils.dataset_manipulation import split_array_with_ratios
+from prescyent.dataset.motion.teleop_icub.config import TeleopIcubDatasetConfig
+
+
+TELEOP_DIMENSIONS_NAMES = [
+    "waist_z", "right_hand_x", "right_hand_y", "right_hand_z",
+    "left_hand_x", "left_hand_y", "left_hand_z"
+]
+
+
+class TeleopIcubDataset(MotionDataset):
+    """TODO: present the dataset here
+    Architecture
+
+    Dataset is not splitted into test / train / val
+    It as to be at initialisation, througt the parameters
+    """
+    def __init__(self, config: Union[Dict, TeleopIcubDatasetConfig] = None,
+                 scaler: Callable = None):
+        if not config:
+            config = TeleopIcubDatasetConfig()
+        self._init_from_config(config)
+        if not Path(self.config.data_path).exists():
+            self._get_from_web()
+        self.episodes = self._load_files()
+        super().__init__(scaler)
+
+    def _init_from_config(self, config):
+        if isinstance(config, dict):
+            config = TeleopIcubDatasetConfig(**config)
+        self.config = config
+        self.input_size = config.input_size
+        self.output_size = config.output_size
+        self.batch_size = config.batch_size
+
+    # load a set of trajectory, keeping them separate
+    def _load_files(self):
+        files = list(Path(self.config.data_path).rglob(self.config.glob_dir))
+        if len(files) == 0:
+            logger.error("No files matching '%s' rule for this path %s",
+                         self.config.glob_dir, self.config.data_path,
+                         group=DATASET)
+            raise FileNotFoundError(self.config.data_path)
+        train_files, test_files, val_files = split_array_with_ratios(files,
+                                                                     self.config.ratio_train,
+                                                                     self.config.ratio_test,
+                                                                     self.config.ratio_val,
+                                                                     shuffle=self.config.shuffle)
+        train_episodes = pathfiles_to_episodes(train_files,
+                                               subsampling_step=self.config.subsampling_step,
+                                               dimensions=self.config.dimensions)
+        test_episodes = pathfiles_to_episodes(test_files,
+                                              subsampling_step=self.config.subsampling_step,
+                                              dimensions=self.config.dimensions)
+        val_episodes = pathfiles_to_episodes(val_files,
+                                             subsampling_step=self.config.subsampling_step,
+                                             dimensions=self.config.dimensions)
+        self.feature_size = train_episodes[0].shape[1]
```

Comment:
We should avoid declaring new attributes outside of the constructor / init functions.